visualization/app/codeCharta/ui/codeMap/threeViewer/threeSceneService.ts

```diff
@@ -53,37 +52,32 @@ export class ThreeSceneService implements CodeMapPreRenderServiceSubscriber {
 		this.reselectBuilding()
 	}
 
-	public highlightBuilding(building: CodeMapBuilding) {
+	public highlightBuildings() {
 		const settings = this.settingsService.getSettings()
-
 		//TODO: Remove once all settings are in the store
 		settings.appSettings.isPresentationMode = this.storeService.getState().appSettings.isPresentationMode
+		this.getMapMesh().setBuildingHighlight(this.listOfBuildingsToHighlight, this.selected, settings)
+	}
 
-		this.getMapMesh().highlightBuilding(building, this.selected, settings)
-		this.highlightedBuildings = []
-		this.highlighted = building
+	public highlightSingleBuilding(building: CodeMapBuilding) {
```

Comment:
If this method is used incorrectly, it might be that you push a building into the highlightedBuildings-Array that already contains something. Let's add `this.clearHighlight` before